common/config/utils.go

```diff
@@ -70,6 +70,66 @@ func parseUnencryptedOtlpGrpcUrl(rawURL string) (grpcUrl string, err error) {
 	return fmt.Sprintf("%s:%s", host, port), nil
 }
 
+// parseEncryptedOtlpGrpcUrl parses and validates a URL for an encrypted OTLP gRPC endpoint.
+// It ensures the scheme is either "https" or "grpcs", and appends the default OTLP gRPC port (4317) if missing.
+func parseEncryptedOtlpGrpcUrl(rawURL string) (grpcUrl string, err error) {
```

Comment:
can we use the same function and pass it an `encrypted bool` parameter?
the implementation is almost identical except for the scheme so can be nice to reuse